pkg/kubelet/kuberuntime/kuberuntime_sandbox.go

```diff
@@ -141,14 +140,11 @@ func (m *kubeGenericRuntimeManager) generatePodSandboxConfig(pod *v1.Pod, attemp
 		return nil, err
 	}
```

Comment:
I have similar desire to refactor this file when I look at it. This PR is taking a very shallow step in refactoring, which makes it a matter of taste as later comment of @bart0sh indicated.

I would suggest to make a little bit bigger effort of separating clearly Windows-only, Linux-only, and shared settings. So when somebody adds a new feature they will not be confused where they put code for it

Reply:
Sure, added a bit more to this PR. Keep in mind that most of this comes from another PR (https://github.com/kubernetes/kubernetes/pull/109761), which was split into 3 (so it's easier to merge, one of them merged, this is still in review: https://github.com/kubernetes/kubernetes/pull/114186), some comments were addressed, a bit extra cleaning.